packages/cli/src/modules/insights/insights.controller.ts

```diff
@@ -1,16 +1,43 @@
-import type { InsightsSummary } from '@n8n/api-types';
+import { ListInsightsWorkflowQueryDto } from '@n8n/api-types';
+import type { InsightsSummary, InsightsByTime, InsightsByWorkflow } from '@n8n/api-types';
 
-import { Get, GlobalScope, RestController } from '@/decorators';
+import { Get, GlobalScope, Query, RestController } from '@/decorators';
+import { paginationListQueryMiddleware } from '@/middlewares/list-query/pagination';
+import { sortByQueryMiddleware } from '@/middlewares/list-query/sort-by';
+import { AuthenticatedRequest } from '@/requests';
 
 import { InsightsService } from './insights.service';
 
 @RestController('/insights')
 export class InsightsController {
+	private readonly nbDaysFilteredInsights = 14;
+
 	constructor(private readonly insightsService: InsightsService) {}
 
 	@Get('/summary')
 	@GlobalScope('insights:list')
 	async getInsightsSummary(): Promise<InsightsSummary> {
 		return await this.insightsService.getInsightsSummary();
 	}
+
+	@Get('/by-workflow', { middlewares: [paginationListQueryMiddleware, sortByQueryMiddleware] })
```

Comment:
Let's add a test making sure the middlewares are used and working as well.

Reply:
I added some api tests here: https://github.com/n8n-io/n8n/pull/14064/files#diff-c2d11772c23ad1280432670d929ea00c9d55b5f3c67bb77b2e72e57b1c1f6c04R45
I found it difficult to put the api tests inside the modules/insight folder (because of lots of dependencies with the shared test folder). But it might make sense to keep integration tests "global" and not module scoped, what da ya think ?

Reply:
Sounds good for now. We have to find another way once we extract them into packages, because the cli should not depend on the module.